pkg/resource/training_job/hooks.go

```diff
@@ -64,4 +69,13 @@ func (rm *resourceManager) customSetOutput(r *resource) {
 	}
 
 	svccommon.SetSyncedCondition(r, trainingJobStatus, &resourceName, &trainingJobModifyingStatuses)
+
+	if ackcompare.IsNil(r.ko.Status.WarmPoolStatus) || ackcompare.IsNil(r.ko.Status.WarmPoolStatus.Status) {
+		return
+	}
+	// Set synced condition to False if Warm Pool is in Inuse or Available state
```

Comment:
logic has been changed from last revision?

since the job has finished I think the decision was to keep resourceSycned as True but continue reconciling until warm pool is terminated and you had another method to implement it, I asked to merge the code but looks like logic itself is different now

Reply:
Yea when I was working on update training job and an error based requeque was causing some issues. But I've refined the code a bit since. Is it fine if we merge this once update training is ready?